addon-test-support/@ember/test-helpers/dom/-is-maxlength-constrained.ts

```diff
@@ -0,0 +1,15 @@
+/**
+  @private
+  @param {Element} element the element to check
+  @returns {boolean} `true` when the element should constrain input by the maxlength attribute, `false` otherwise
+*/
+export default function isMaxLengthConstrained(element: Element) {
```

Comment:
New to TS, had a couple types questions here.
I noticed `-is-form-control` had an export type FormControl. Does a type guard make sense here?
```typescript
import { FormControl } from './-isFormControl';

export default function isMaxLengthConstrained(element: Element) element is FormControl {
```

Also I noticed formControl had the parameter annotation `isFormControl(element: Element | Document)` and I wasn't sure what the Document bit was about

Reply:
I don't think `-is-form-control` is applicable here, cause in addition to inputs and textareas it does also account for buttons and selects, which, I think, is un-desirable. However, you could add you own type guard here to specify `element` type on the invocation side, like:
```ts
export function isMaxLengthConstrained(element: Element): element is HTMLInputElement | HTMLTextAreaElement {
// ...
```

> isFormControl(element: Element | Document)

ya, I believe it should not accept `Document`. I suspect it's encouraged by `getElement` which is currently able to return `Document` https://github.com/emberjs/ember-test-helpers/blob/637fdbc8a45cfed013e47701c786d19c89cf06ad/addon-test-support/%40ember/test-helpers/dom/-get-element.ts#L16

and we use its result to pass it down to the other interaction helpers. I think `getElement` return types may be worth to reconsider. But this is probably unrelated to this PR.